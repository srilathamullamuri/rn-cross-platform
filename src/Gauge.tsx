/***********************************************************************EMCL2022
 *
 * Gauge.tsx
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of EMC Software and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to EMC Software
 * and its suppliers and may be covered by U.S. and Foreign Patents,
 * patents in process, and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from EMC Software.
 *
 * @flow
 *
 * EMC Software Confidential, All Rights Reserved
 * Copyright (c) 2022, EMC Software
 *
 ***************************************************************************/

import * as React from 'react';
import {View, Text} from 'react-native';
import ReactTooltip from 'react-tooltip';
import {colors} from './styles';

export interface GProps {
  label: string;
  color: string;
  pct: number;
  val: number;
  max: number;
  target?: number;
}

export const Gauge = (props: {meters: GProps[]}) => {
  return (
    <>
      {props.meters?.map((item, index) => {
        // console.log('item', item);
        let lowMark;
        let highMark;
        let optMark;

        // crazy rules for changing color - see https://css-tricks.com/html5-meter-element/
        if (item.color === 'red') {
          lowMark = item.pct + 1;
          highMark = item.pct + 1;
          optMark = item.pct + 2;
        } else if (item.color === 'yellow') {
          lowMark = item.pct + 1;
          highMark = item.pct + 1;
          optMark = item.pct + 1;
        } else if (item.color === 'green') {
          lowMark = item.pct - 1;
          highMark = item.pct + 1;
          optMark = item.pct;
        }

        const helpText = `Value is ${item.val} out of ${item.max}. Target is ${item.target}`;
        const targetString = `(LR => ${item.target || item.max})`;

        return (
          <View
            key={index}
            style={{
              flex: 1,
              backgroundColor: colors.ochBlueP3,
              flexDirection: 'column',
              alignItems: 'center',
              padding: 0,
              margin: 10,
              borderRadius: 10,
            }}>
            <ReactTooltip />
            <Text
              style={{
                padding: 5,
                borderTopEndRadius: 50,
                borderBottomEndRadius: 50,
                fontWeight: 'bold',
              }}>
              {item.label}
            </Text>
            <meter
              id={item.label.replaceAll(' ', '')}
              min="0"
              data-tip={helpText}
              low={lowMark}
              color={item.color}
              optimum={optMark}
              high={highMark}
              max="100"
              style={{
                display: 'block',
                height: '50px',
              }}
              value={item.pct}
            />
            <View style={{flexDirection: 'row'}}>
              <Text
                style={{
                  padding: 5,
                  borderTopEndRadius: 50,
                  borderBottomEndRadius: 50,
                  fontWeight: 'bold',
                }}>
                {item.val}
              </Text>
              <Text
                style={{
                  padding: 5,
                  borderTopEndRadius: 50,
                  borderBottomEndRadius: 50,
                }}>
                {targetString}
              </Text>
            </View>
          </View>
        );
      })}
    </>
  );
};
