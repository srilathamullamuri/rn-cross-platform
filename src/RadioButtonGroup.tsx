/***********************************************************************EMCL2022
 *
 * RadioButtonGroup.tsx
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of EMC Software and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to EMC Software
 * and its suppliers and may be covered by U.S. and Foreign Patents,
 * patents in process, and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from EMC Software.
 *
 * @flow
 *
 * EMC Software Confidential, All Rights Reserved
 * Copyright (c) 2022, EMC Software
 *
 ***************************************************************************/

import React, {useState} from 'react';
import {View, TouchableOpacity} from 'react-native';

export const RadioButtonGroup = (props: {
  style?: any;
  data: any;
  value: string;
  columnWidth: number;
  onPress: (event) => void;
  editable: boolean;
}) => {
  const [value, setValue] = useState<string>(props.value);

  React.useEffect(() => {
    setValue(props.value);
  }, [props.value]);

  return (
    <View style={{width: '100%', flexDirection: 'row', flex: 1}}>
      {props.data.map((item, index) => {
        const isSelected = item.id === value;
        return (
          <View
            key={index}
            style={{
              width: props.columnWidth + '%',
              flex: 1,
              justifyContent: 'center',
              alignItems: 'center',
            }}>
            <TouchableOpacity
              key={index}
              onPress={() => {
                if (props.editable) {
                  setValue(item.id);
                  props.onPress(item.id);
                }
              }}
              style={[
                {
                  height: 24,
                  width: 24,
                  borderRadius: 12,
                  borderWidth: 2,
                  borderColor: '#000',
                  alignItems: 'center',
                  justifyContent: 'center',
                },
                props.style,
              ]}>
              {isSelected ? (
                <View
                  style={{
                    height: 12,
                    width: 12,
                    borderRadius: 6,
                    backgroundColor: '#000',
                  }}
                />
              ) : null}
            </TouchableOpacity>
          </View>
        );
      })}
    </View>
  );
};
