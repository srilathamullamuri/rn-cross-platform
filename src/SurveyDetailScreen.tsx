/***********************************************************************EMCL2022
 *
 * SurveyDetailScreen.tsx
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of EMC Software and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to EMC Software
 * and its suppliers and may be covered by U.S. and Foreign Patents,
 * patents in process, and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from EMC Software.
 *
 * @flow
 *
 * EMC Software Confidential, All Rights Reserved
 * Copyright (c) 2022, EMC Software
 *
 ***************************************************************************/

import * as React from 'react';
import {View, Text} from 'react-native';

// import {View, Text, StyleSheet, StatusBar, FlatList} from 'react-native';
import SurveyWidget from './SurveyWidget';
import {colors,styles} from './styles';
import {Gauge} from './Gauge';
import {SurveyBottomNavBar} from './SurveyBottomNavBar';
import {
  SurveyAnswerDict,
  SurveyScoreDict,
  SurveyType,
  SurveyDictOfStrings,
  SurveyTreeNode,
  SurveyAnswerSetType,
} from './SurveyTypes';
import {SurveyTabList} from './SurveyTabList';
import {ModalFile} from './components/ModalFile';
export interface IProps {
  targetUser: any;
  survey: any;
  sendSurveyRes: (
    item: any,
    value: string,
    status: string,
    surveyAnswerSet: any,
  ) => void;
  selectedSurvey: (item: any) => void;
  showQuestions: boolean;
  surveyAnswer: SurveyAnswerDict;
  scores: any; // SurveyScoreDict; < compile issue
  tabScores: SurveyScoreDict;
  surveyStatus: SurveyDictOfStrings;
  surveyAnswerSet?: SurveyAnswerSetType;
  freezeAnswerSet: (surveyAnswerSetId: string) => void;
  viewPdf:(item)=> void;
  modalFileUrl: string | undefined;
}

// Layout of survey screen - banner, navigation and questions.
export const SurveyDetailScreen = (props: IProps) => {
  // this is the entire survey definition
  const [survey, setSurvey] = React.useState<SurveyType>([]);
  const [surveyTree, setSurveyTree] = React.useState<SurveyTreeNode[]>([]);
  const [surveyAnswer, setSurveyAnswer] = React.useState<SurveyAnswerDict>();
  const [surveyAnswerSet, setSurveyAnswerSet] = React.useState<
    SurveyAnswerSetType | undefined
  >(props.surveyAnswerSet);
  const [unsavedAnswers, setUnsavedAnswers] = React.useState<SurveyAnswerDict>(
    {},
  );
  const [sortedDisplay, setSortedDisplay] = React.useState<number[]>([]);

  const [selectedTabIdx, setSelectedTabIdx] = React.useState<number>(0);

  // leaving these 6 warnings in place for now - will need to use shortly
  // const [showQuestions, setShowQuestions] = React.useState<boolean>(
  //   props.showQuestions,
  // );
  // const [showAnswer, setShowAnswer] = React.useState<boolean>(false);
  const {survey: props_survey, selectedSurvey: props_selectedSurvey} = props;
  const [canSave, setCanSave] = React.useState<boolean>(false);
  const [showPdf, setShowPdf] = React.useState<boolean>(false);
  const {modalFileUrl} = props;
  React.useEffect(() => {
    // determine if first run,  survey state will be empty
    const isFirstRun = survey.length === 0;
    console.log('etg74 - ', {isFirstRun});

    if (isFirstRun) {
      // preprocess json string. Add index and parse value
      const surveyFromProps = props_survey;
      // props_survey.map((i, idx) => {
      //   i.parsed = JSON.parse(i.value);
      //   i.idx = idx;
      //   return i;
      // });

      // extract meta data for manipulation into ordered hierarchical array
      const flatDir: SurveyTreeNode[] = surveyFromProps.map((item, idx) => {
        // console.log({value});
        return {
          topName: item.tabName.replace(/\..*$/, ''),
          name: item.tabName,
          order: item.value?.order || 99,
          idx,
        };
      });
      const flatDirSorted = flatDir.sort((a, b) => {
        if (a.topName === b.topName) {
          return a.order - b.order;
        }
        const diff = a.order - b.order;
        if (diff === 0 || (a.order === 99 && b.order === 99)) {
          if (a.name < b.name) {
            return -1;
          }
          if (a.name > b.name) {
            return 1;
          }
          return 0;
        }
        return diff;
      });
      console.log({flatDirSorted});

      // make l an array of unique top level tab names
      // const topLevelNames = [...new Set(flatDirSorted.map((i) => i.topName))];

      // build tree structure.
      const surveyTree: SurveyTreeNode[] = [];
      flatDirSorted.forEach((item) => {
        const newNode = (node: SurveyTreeNode, top = false) => {
          if (node.topName === node.name || !top) {
            return node;
          }
          return {
            ...node,
            name: node.topName,
            branches: [node],
          };
        };

        const lastItemIdx = surveyTree.length - 1;
        if (lastItemIdx >= 0) {
          if (item.name !== item.topName) {
            const matchingIdx = surveyTree.findIndex(
              (i) => i.topName === item.topName,
            );
            if (matchingIdx >= 0) {
              surveyTree[matchingIdx].branches?.push(newNode(item));
            } else {
              console.log({
                name: item.name,
                topName: item.topName,
                lastItemIdx,
              });
              surveyTree.push(newNode(item, true));
            }
          } else {
            console.log({
              name: item.name,
              topName: item.topName,
              lastItemIdx,
            });
            surveyTree.push(newNode(item, true));
          }
        } else {
          console.log({
            name: item.name,
            topName: item.topName,
            lastItemIdx,
          });
          surveyTree.push(newNode(item, true));
        }
      });
      console.log({surveyTree});

      const selectedTab = flatDirSorted[0].idx;
      const selectedTabName = surveyFromProps[selectedTab].tabName;
      const selectedTabRowId = surveyFromProps[selectedTab].id;
      console.log({surveyTree, selectedTab, selectedTabName});

      // Display order.. as flat array
      const sortedDisplay: number[] = [];
      surveyTree.forEach((topItem) => {
        if (topItem.branches?.length) {
          topItem.expanded = false;
          topItem.branches.forEach((botItem) => {
            sortedDisplay.push(botItem.idx);
          });
        } else {
          sortedDisplay.push(topItem.idx);
        }
      });

      // record sorted order in surveyFromProps
      sortedDisplay.forEach((i, idxSorted) => {
        surveyFromProps[i].idxSorted = idxSorted;
      });
      // console.log('POST idxSorted ', {surveyFromProps});

      setSurvey(surveyFromProps);
      setSelectedTabIdx(selectedTab);
      setSortedDisplay(sortedDisplay);
      setSurveyTree(surveyTree);

      // get answers!
      props_selectedSurvey(selectedTabRowId);
    }
  }, [props_selectedSurvey, props_survey, survey.length]);

  // React.useEffect(() => {
  //   setShowQuestions(props.showQuestions);
  // }, [props.showQuestions]);

  React.useEffect(() => {
    setSurveyAnswerSet(props.surveyAnswerSet);
  }, [props.surveyAnswerSet]);

  React.useEffect(() => {
    setSurveyAnswer(props.surveyAnswer);
  }, [props.surveyAnswer]);

  /**
   * show tab identified by idx
   * @param idx
   */
  const showTab = (idx: number) => {
    setShowPdf(false);
    if (idx < 0 || typeof idx === 'undefined') {
      return; // maybe go up a level.
    }
    const item = survey[idx];
    console.log('showTab 200', {idx, item});
    props_selectedSurvey(item.id); // should be by DB row id

    // Done above
    // if (!item.parsed) {
    //   item.parsed = JSON.parse(item.value);
    //   if (!item.parsed) {
    //     console.log('ERROR PARSING QUESTIONS: ', {value: item.value});
    //   }
    // }

    setSelectedTabIdx(idx);
    setCanSave(false);
  };
  const viewReports = (item) => {
    if(item){
      setShowPdf(true);
      props.viewPdf(surveyAnswerSet?.id);
    }else{
      setShowPdf(false);
    }
    
  }
  const sendSurveyRes = (item, status) => {
    const selectedTab = survey[selectedTabIdx].id;
    setUnsavedAnswers(item);
    console.log('sendSurveyRes', {item, selectedTab, status});
    props.sendSurveyRes(item, selectedTab, status, surveyAnswerSet);
    const ansTmp: SurveyAnswerDict = {};
    setUnsavedAnswers(ansTmp);
  };

  const saveAnswers = () => {
    // Simple - does q count = a count?
    sendSurveyRes(unsavedAnswers, 'Started');
    setCanSave(false);
  };
  const keepUnsavedAnswers = (a) => {
    setUnsavedAnswers(a);
    setCanSave(true);
  };
  const finalizeSurveySet = () => {
    // setUnsavedAnswers(a);
    surveyAnswerSet && props.freezeAnswerSet(surveyAnswerSet.id);
  };

  const selectedSurveyTab = survey[selectedTabIdx]?.value;
  const sortedIdx =
    1 + (survey[selectedTabIdx]?.idxSorted ?? sortedDisplay.length);
  const nextItem =
    (sortedIdx < sortedDisplay.length && sortedDisplay[sortedIdx]) as number ?? -1;

  return (
    <View
      style={{
        width: '100%',
        flex: 1,
        flexDirection: 'row',
        flexWrap: 'wrap',
        alignItems: 'flex-start',
      }}>
      <View style={{flex: 1, flexDirection: 'column', minWidth: '180px'}}>
        <View>
          <Text
            style={{
              textTransform: 'capitalize',
              fontWeight: 'bold',
            }}>
            {props.targetUser}
          </Text>
        </View>
        <SurveyTabList
          surveyTree={surveyTree}
          tabScores={props.tabScores}
          showTab={showTab}
          selectedTabIdx={selectedTabIdx}
          surveyStatus={props.surveyStatus}
          viewReport={viewReports}
        />
      </View>
      <View style={{flex: 6}}>
      {!showPdf && (<View>
        <View
          style={{
            width: '100%',
            padding: 5,
            borderColor: colors.black,
            flexDirection: 'row',
          }}>
          <Gauge meters={props.scores} />
        </View>
        <SurveyWidget
          surveyAnswer={surveyAnswer || {}}
          surveyTab={selectedSurveyTab?.survey}
          sendSurveyRes={(item, status) => sendSurveyRes(item, status)}
          onAnswer={keepUnsavedAnswers}
          viewMode={surveyAnswerSet?.bFinal}
        />
        <SurveyBottomNavBar
          next={nextItem}
          onFinal={finalizeSurveySet}
          onSave={saveAnswers}
          onNav={showTab}
          canSave={canSave}
        />
      </View>)}
      
      {showPdf && modalFileUrl && (
        <ModalFile
          url={modalFileUrl}
          onXClick={() => setShowPdf(false)}
          styles={{
            modStyle: styles.docCustomModelStyles,
            jpgStyle: styles.imagePDF,
          }}
          viewFileType="pdf"
        />
      )}</View>
    </View>
  );
};
