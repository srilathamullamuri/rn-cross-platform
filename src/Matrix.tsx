/***********************************************************************EMCL2022
 *
 * Matrix.tsx
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of EMC Software and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to EMC Software
 * and its suppliers and may be covered by U.S. and Foreign Patents,
 * patents in process, and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from EMC Software.
 *
 * @flow
 *
 * EMC Software Confidential, All Rights Reserved
 * Copyright (c) 2022, EMC Software
 *
 ***************************************************************************/

import React from 'react';
import {Text, TextStyle, View, ViewStyle} from 'react-native';

import {colors} from './styles';
import {InfoButton} from './components/InfoButton';
import {RadioButtonGroup} from './RadioButtonGroup';
import {SurveyAnswerDict} from './SurveyTypes';

export const Matrix = (props: {
  question: any;
  styles: {[k: string]: ViewStyle | TextStyle};
  answers: SurveyAnswerDict;
  updateAnswers: (ansDict: SurveyAnswerDict, dbg?: string) => void;
  disableColumnSet?: boolean;
  viewMode: boolean;
}) => {
  // const [value, setValue] = useState<string>(props.value);
  const {answers, question, updateAnswers, disableColumnSet = false} = props;
  const {rows, columns, title, questionId, info} = question;
  const {questionTextStyle, containerStyle} = props.styles;

  React.useEffect(() => {
    console.log('selectedRenderMatrix', question);
    console.log('selectedRenderMatrix 608', {
      rows,
      columns,
      title,
      questionId,
      // tabAnswerDict,
    });
    console.log('etg622', {answers});
  }, [answers, columns, props, question, questionId, rows, title]);

  const columnWidth = 100 / (columns.length + 2);
  const rowTitleWidth = columnWidth * 2;

  return (
    <View
      key={`matrix-${questionId}`}
      style={[
        containerStyle,
        {width: '100%', backgroundColor: '#FFF', padding: '5%'},
      ]}>
      <View style={{flexDirection: 'row'}}>
        <InfoButton iconColor={colors.ochBlue} text={info} />
        <Text style={questionTextStyle}>{title}</Text>
      </View>
      <View
        style={{
          flex: 1,
          flexDirection: 'row',
          justifyContent: 'space-between',
          borderBottomColor: '#DDD',
          borderBottomWidth: 1,
          paddingBottom: 10,
          paddingTop: 10,
          alignItems: 'flex-end',
        }}>
        <InfoButton iconColor={colors.white} text={info} />
        <View
          style={{
            width: `${rowTitleWidth}%`,
          }}></View>
        {columns.map((colItem, index) => {
          return (
            <View
              key={`colMap-${index}`}
              style={{
                width: `${columnWidth}%`,
                flex: 1,
                justifyContent: 'center',
                alignItems: 'center',
              }}>
              <Text
                style={{
                  fontWeight: 'bold',
                  textAlign: 'center',
                  // fontSize: 20,
                }}>
                {colItem.optionText}
              </Text>
              {!disableColumnSet && rows.length > 1 && (
                <input
                  style={{
                    zoom: 1.5,
                  }}
                  checked={false}
                  type="checkbox"
                  onChange={() => {
                    console.log('onChange -');
                    if (!props.viewMode) {
                      const a: SurveyAnswerDict = {};
                      rows.forEach((rowItem, _index) => {
                        console.log('Check ', {rowItem, colItem});
                        a[rowItem.id] = colItem.id;
                      });
                      updateAnswers(a, '107');
                      console.log('Check ');
                    }
                  }}
                />
              )}
            </View>
          );
        })}
      </View>
      {rows.map((item, index) => {
        console.log('rows.map 653', {
          item,
          index,
          answers,
          value: answers[item.id] ?? '',
          columns,
        });
        const value: string = (answers[item.id] ?? '') as string;
        return (
          <View
            style={{
              flex: 1,
              flexDirection: 'row',
              justifyContent: 'space-between',
              borderBottomColor: '#DDD',
              borderBottomWidth: 1,
              paddingBottom: 10,
              paddingTop: 10,
            }}
            key={`rMap-${index}`}>
            <InfoButton iconColor={colors.ochBlue} text={item.info} />
            <View
              style={{
                width: `${rowTitleWidth}%`,
              }}>
              <Text style={{fontWeight: 'bold'}}>{item.optionText}</Text>
            </View>
            <RadioButtonGroup
              value={value}
              data={columns}
              columnWidth={columnWidth}
              editable={!props.viewMode}
              onPress={(answerId) => {
                const a: SurveyAnswerDict = {};
                a[item.id] = answerId;
                updateAnswers(a, '698');
              }}></RadioButtonGroup>
          </View>
        );
      })}
    </View>
  );
};
