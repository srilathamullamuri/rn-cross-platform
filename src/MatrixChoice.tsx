/***********************************************************************EMCL2022
 *
 * MatrixChoice.tsx
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of EMC Software and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to EMC Software
 * and its suppliers and may be covered by U.S. and Foreign Patents,
 * patents in process, and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from EMC Software.
 *
 * @flow
 *
 * EMC Software Confidential, All Rights Reserved
 * Copyright (c) 2022, EMC Software
 *
 ***************************************************************************/

import React from 'react';
import {TextStyle, ViewStyle} from 'react-native';
import {SurveyAnswerDict} from './SurveyTypes';
import {Matrix} from './Matrix';

/* just like the matrix except only one choice can be made */
export const MatrixChoice = (props: {
  question: any;
  styles: {[k: string]: ViewStyle | TextStyle};
  answers: SurveyAnswerDict;
  updateAnswers: (ansDict: SurveyAnswerDict, dbg?: string) => void;
  viewMode: boolean;
}) => {
  // const [value, setValue] = useState<string>(props.value);
  const {answers, question, styles, updateAnswers} = props;
  const {rows, columns, title, questionId} = question;
  // const {questionTextStyle, containerStyle} = props.styles;
  const [tabAnswerDict, setTabAnswerDict] = React.useState<SurveyAnswerDict>(
    {},
  );

  React.useEffect(() => {
    console.log('selectedRenderMatrix', question);
    console.log('selectedRenderMatrix 608', {
      rows,
      columns,
      title,
      questionId,
      // tabAnswerDict,
    });
    console.log('etg622', {answers});
  }, [answers, columns, props, question, questionId, rows, title]);

  React.useEffect(() => {
    // only pass the answer that matches questionId after parsing and rebuilding.
    const a = {};
    const words = (answers[questionId] as string)?.split(':');
    if (words?.length === 2) {
      a[words[0]] = words[1];
      setTabAnswerDict(a);
    }
  }, [answers, questionId]);

  // const columnWidth = 100 / (columns.length + 2);
  // const rowTitleWidth = columnWidth * 2;

  const updateAnswersIntercept = (ansDict: SurveyAnswerDict, dbg?: string) => {
    console.log('updateAnswersIntercept', {ansDict});
    // rebuild answer as answer of the entire question
    const [key, value] = Object.entries(ansDict)?.[0];
    const a = {};
    a[questionId] = `${key}:${value}`;
    updateAnswers(a, dbg);
  };

  return (
    <Matrix
      question={question}
      answers={tabAnswerDict}
      styles={styles}
      updateAnswers={updateAnswersIntercept}
      disableColumnSet={true}
      viewMode={props.viewMode}
    />
  );
};
