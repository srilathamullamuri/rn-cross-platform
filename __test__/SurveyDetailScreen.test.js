import React from 'react';
import renderer from 'react-test-renderer';
import {SurveyDetailScreen} from '../src/SurveyDetailScreen';

test('renders correctly survey detail screen', () => {
  const tree = renderer.create(<SurveyDetailScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
