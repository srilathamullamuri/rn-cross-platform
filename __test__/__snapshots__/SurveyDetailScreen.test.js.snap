// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly survey detail screen 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "minWidth": "180px",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "textTransform": "capitalize",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <View>
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={0.5}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "margin": "4.25rem",
          }
        }
      />
      <View>
        <View
          style={
            Object {
              "backgroundColor": "",
              "borderBottomEndRadius": 50,
              "borderTopEndRadius": 50,
              "flex": 1,
              "flexDirection": "row",
              "padding": 5,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            +
          </Text>
          <Text
            style={
              Object {
                "flex": 8,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
               
               REPORTS
            </Text>
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginLeft": 15,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 6,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "borderColor": "#000000",
            "flexDirection": "row",
            "padding": 5,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "space-around",
                "alignItems": "stretch",
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "minHeight": 300,
          }
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
              "marginBottom": 10,
              "marginTop": 10,
              "maxWidth": 100,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                disabled={false}
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "green",
                    },
                  ]
                }
              >
                Previous
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexGrow": 1,
              "marginBottom": 10,
              "marginTop": 10,
              "maxWidth": 100,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                disabled={false}
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "green",
                    },
                  ]
                }
              >
                Complete / Submit
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexGrow": 1,
              "marginBottom": 10,
              "marginTop": 10,
              "maxWidth": 100,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                  Object {},
                ]
              }
            >
              <Text
                disabled={true}
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "green",
                    },
                    Object {
                      "color": "#cdcdcd",
                    },
                  ]
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexGrow": 1,
              "marginBottom": 10,
              "marginTop": 10,
              "maxWidth": 100,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                disabled={false}
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "green",
                    },
                  ]
                }
              >
                Next
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
